graphrag/query/cli.py

```diff
@@ -177,12 +180,13 @@ def _configure_paths_and_settings(
     root_dir: str | None,
     config_filepath: str | None,
 ) -> tuple[str, str | None, GraphRagConfig]:
+    config = _create_graphrag_config(root_dir, config_filepath)
     if data_dir is None and root_dir is None:
         msg = "Either data_dir or root_dir must be provided."
         raise ValueError(msg)
     if data_dir is None:
-        data_dir = _infer_data_dir(cast(str, root_dir))
-    config = _create_graphrag_config(root_dir, config_filepath)
+        base_dir = Path(cast(str, root_dir)) / config.storage.base_dir
+        data_dir = cast(str, resolve_timestamp_path(base_dir))
```

Comment:
Instead of cast you can do str(resolve_timestamp_path(base_dir))
Generally cast is used to tell the type checker to treat the item as str, whereas str() actually converts the itme.